src/main/scala/fi/oph/koski/schema/VapaaSivistystyöKotoutumiskoulutus2022.scala

```diff
@@ -0,0 +1,223 @@
+import fi.oph.scalaschema.annotation.{Description, OnlyWhen, Title}
+
+import java.time.LocalDate
+
+/*
+  Päätason suoritus
+*/
+
+@OnlyWhen("koulutusmoduuli/perusteenDiaarinumero", "OPH-XXX-2022") // TODO: Oikea diaarinumero
+case class OppivelvollisilleSuunnattuMaahanmuuttajienKotoutumiskoulutuksenSuoritus2022(
+  toimipiste: OrganisaatioWithOid,
+  @KoodistoKoodiarvo("vstmaahanmuuttajienkotoutumiskoulutus")
+  tyyppi: Koodistokoodiviite = Koodistokoodiviite(koodiarvo = "vstmaahanmuuttajienkotoutumiskoulutus", koodistoUri = "suorituksentyyppi"),
+  koulutusmoduuli: VSTKotoutumiskoulutus2022 = VSTKotoutumiskoulutus2022(),
+  vahvistus: Option[HenkilövahvistusValinnaisellaPaikkakunnalla] = None,
+  @Description("Koulutuksen opetuskieli")
+  @Title("Opetuskieli")
+  suorituskieli: Koodistokoodiviite,
+  @Title("Osaamiskokonaisuudet")
+  override val osasuoritukset: Option[List[VSTKotoutumiskoulutuksenKokonaisuudenSuoritus2022]] = None,
+  @Description("Todistuksella näytettävä lisätieto, vapaamuotoinen tekstikenttä")
+  todistuksellaNäkyvätLisätiedot: Option[LocalizedString] = None
+) extends VapaanSivistystyönPäätasonSuoritus
+  with SuoritusVaatiiMahdollisestiMaksuttomuusTiedonOpiskeluoikeudelta
+  with OpintopistelaajuuksienYhteislaskennallinenPäätasonSuoritus
+  with OpintopistelaajuuksienYhteislaskennallinenSuoritus
+
+@Description("Vapaan sivistystyön maahanmuuttajien kotoutumiskoulutuksen (OPS 2022) tunnistetiedot")
+case class VSTKotoutumiskoulutus2022(
+  @KoodistoKoodiarvo("999910") // TODO: Säilyykö tämä samana? 999910 = Kotoutumiskoulutus oppivelvollisille
+  tunniste: Koodistokoodiviite = Koodistokoodiviite("999910", koodistoUri = "koulutus"),
+  perusteenDiaarinumero: Option[String] = None,
+  koulutustyyppi: Option[Koodistokoodiviite] = None,
+  laajuus: Option[LaajuusOpintopisteissä] = None
+) extends DiaarinumerollinenKoulutus with Tutkinto with OpintopistelaajuuksienYhteenlaskennallinenKoulutusmoduuli
+
+@Title("Vapaan sivistystyön kotoutumiskoulutuksen arviointi")
+@Description("Hyväksytty arviointi tarkoittaa osasuorituksen suoritusta")
+case class VSTKotoutumiskoulutuksenOsasuorituksenArviointi2022(
+  @KoodistoKoodiarvo("Hyväksytty")
+  arvosana: Koodistokoodiviite = Koodistokoodiviite("Hyväksytty", "arviointiasteikkovst"),
+  päivä: LocalDate
+) extends ArviointiPäivämäärällä with VapaanSivistystyönKoulutuksenArviointi
+
+/*
+  Osasuoritukset
+*/
+
+@Description("Vapaan sivistystyön maahanmuuttajien kotoutumiskoulutuksen osasuoritus")
+trait VSTKotoutumiskoulutuksenKokonaisuudenSuoritus2022 extends Suoritus with Vahvistukseton with OpintopistelaajuuksienYhteislaskennallinenSuoritus {
+  override val arviointi: Option[List[VSTKotoutumiskoulutuksenOsasuorituksenArviointi2022]]
+  @KoodistoUri("suorituksentyyppi")
+  override val tyyppi: Koodistokoodiviite
+}
+
+@Description("Vapaan sivistystyön maahanmuuttajien kotoutumiskoulutuksen osasuorituksen osasuoritus")
```

Comment:
Pohdiskelen tässä näitä termejä, että olisiko olemassa jokin domainista tuleva termi "osasuorituksen osasuoritukselle", joka olisi skeeman käyttäjälle kuvaavampi kuin tämä?

Alempana puhutaan myös "VST-KOTO kieli- ja viestintäosaamisen osasuorituksen alaosasuoritus" -termillä, voisi olla ainakin hyvä käyttää tai olla käyttämättä "ala-"prefiksiä konsistentisti.

Reply:
Nyt kun tarkemmin katson: Miksi tässä tietomallissa on nämä alaosasuoritukset ollenkaan, kun näyttäisi siltä, että niihin on sallittua tallentaa alaosasuorituksia vain tasan yhdellä  koulutusmoduulin tunnisteella, eikä ole edes mitään vapaatekstikenttää, jossa voisi spesifioida, mitä tarkkaanottaen on opiskeltu?

Eikö tämä tee alaosasuorituksista vain redundanttia dataa?

Reply:
Osittain sekoilua, jos nyt oikein katsoin: alaosasuoritukset onkin tyypitetty alaosasuorituksen tyyppi-kentällä, ja koulutusmoduulin tunniste on kaikissa sama. Mielestäni tämä on vähän epäloogista, ja koulutuksen sisältöön perustuva ryhmittely pitäisi tehdä koulutusmoduulin tunnisteilla eikä tyyppi-kentällä. Niin mielestäni ainakin muissa tietomalleissa on tyypillisesti tehty.

Tietty, jos VST:ssä on näin tehty jo edellisellä perusteella, niin konsistenttius sen kanssa lienee kuitenkin enemmän oikein...

Reply:
Koulutusmoduulin kanssa tosiaan käynyt tilanne, jossa on vapautunut pahanhajuista kaasua aivoista. Se on nyt sorvattu toivottavasti järkevämpään suuntaan. Valitettavasti domainista emme tiedä vielä järkevää termiä...